src/software/amazon/ion/impl/lite/IonLoaderLite.java

```diff
@@ -122,21 +122,34 @@ public IonDatagram load(Reader ionText) throws IonException, IOException
 
     public IonDatagram load(byte[] ionData) throws IonException
     {
+        IonReader reader = null;
         try {
-            IonReader reader = makeReader(_catalog, ionData, 0, ionData.length, _lstFactory);
+            reader = makeReader(_catalog, ionData, 0, ionData.length, _lstFactory);
             IonDatagramLite datagram = load_helper(reader);
             return datagram;
         }
         catch (IOException e) {
             throw new IonException(e);
         }
+        finally
```

Comment:
All supported versions of Java support try with resources. Java 6 and earlier have fallen out of the support window from Oracle as of 12/2018.

Are there any known consumers that require Java 6 compatibility?

Reply:
Yes, there are. Sorry, but I can't give more detail than that.